### DISCUSSION TITLE
Removing unused controls in HID

### DISCUSSION THREAD

**Discussion by yaboirus** [NicoHood/HID#408] [Q&A]
Hi! I am currently working on a project that is similar to a button box. is it possible to remove unwanted controls like unused buttons and unused axis? If so what code would I have to write to make this possible?

**Comment by NicoHood** [comment]
That would not help in any case. What benefit do you expect? the code will be mostly the same size anyways

**Reply by yaboirus** [reply]
Hi thank you for replying! I think the main benefit of removing unused buttons and axis is that it would make a much cleaner interface when binding controls. I use it in a flight sim like x-plane. It keeps reminding me to calibrate non existing axis. Maybe there is a code that could exclude unused controls like in the joystick library? Again thank you very much!

**Reply by WRusman** [reply]
I would also like to be able to configure the number of buttons and axes for example, Not all joysticks use all the buttons and axes.
Seeing 32 buttons, 4 axes and a top-hat on a device with only1 axis and 4 physical buttons will confuse end-users.
Enabling only the connected buttons will make it easier for the end-user to choose the axes and buttons to use in the pc software.

**Reply by xgpt** [reply]
This is a fair use case for this screenshot. How hard would it be to add something that's like "disable button 6" etc?
